platform/linux-generic/pool/subsystem.c

```diff
@@ -15,3 +19,219 @@ ODP_SUBSYSTEM_CONSTRUCTOR(pool)
 	odp_subsystem_constructor(pool);
 }
 
+int odp_pool_init_global(void)
+{
+	pool_module_t *mod;
+
+	mod = odp_subsystem_active_module(pool, mod);
+	if (mod == NULL) {
+		ODP_ERR("No active module in pool subsystem\n");
+		return -1;
+	}
+	if (mod->base.init_global == NULL) {
+		ODP_ERR("No defined init_global function "
+			"in module %s of pool subsystem\n", mod->base.name);
+		return -1;
+	}
```

Comment:
This pattern is highly repetitive, which suggests it should be factored out as part of the modular framework itself rather than having to be repeated for every API. I fact it seems that the entire subsystem.c file for each module could be auto-generated since there's no real variant code here.